### ISSUE TITLE
Second argument to `Annotated` should not be assumed to be a valid type annotation

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1163]
**Describe the bug**
The second argument to `Annotated` should not be assumed to be a valid type annotation, e.g. a generic if subscripted.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That it would produce no errors.

**Screenshots or Code**
```python
from __future__ import annotations

from typing_extensions import Annotated


class Foo:
    def __class_getitem__(cls, args: tuple[int, ...]) -> None:
        ...

# Expected class type but received "Literal[10]"
# Expected no type arguments
# Expected class type but received "Literal[63]"
bar: Annotated[str, Foo[10, 63]] = ''

```

**VS Code extension or command-line**
Either, 1.1.85

**Comment by erictraut** [comment]
This problem has nothing to do with `Annotated`. It's related instead to lack of support for the custom `__class_getitem__` magic method to override behavior for class subscripting. I've added basic support for this to eliminate the error. This support doesn't verify that the subscript arguments are the correct type. I think that use of this magic method will be extremely rare, so I'd prefer not to add more complexity and overhead to support type checking its arguments. If it gains popularity at some point, this could be added.

**Comment by layday** [comment]
Ah, I thought I observed the same behaviour with `__getitem__` but I was wrong.  Thanks for explaining.

**Comment by erictraut** [comment]
This is now addressed in 1.1.86, which I just published. It will also be included in the next release of pylance.
